addons/sale/models/account_move_line.py

```diff
@@ -112,6 +116,9 @@ def _sale_create_reinvoice_sale_line(self):
                         # store it in the map_move_sale_line map
                         map_move_sale_line[move_line.id] = len(sale_line_values_to_create) - 1  # save the index of the value to create sale line
 
+            elif move_line.move_id.move_type in ('out_refund', 'in_refund'):
```

Comment:
Don't we want to create the SO line too, like above if we didn't find a matching one?

Reply:
I would say no? 
I cannot think of a scenario in which `move_line.move_id.move_type in ('out_refund', 'in_refund')` and there are no so_line in `_sale_create_reinvoice_sale_line`

Reply:
But you also need some caching etc. using `map_entry_key` like above, don't you?